### DISCUSSION TITLE
Elements hover and link tool-the arrows on all sides appear like in draw.io

### DISCUSSION THREAD

**Discussion by Naveena1321** [clientIO/joint#2782] [Q&A] [answered]
### Introduction

![Screenshot+from+2024-10-28+11-17-46](https://github.com/user-attachments/assets/ee126171-f6c2-4fcb-8924-caa8c2413483)
is there a way to do something like this(i mean connection points and hovered link tool-the arrows on all sides on hover) in jointjs with stencills and paper and inpector setup

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment]
Have you seen this example? https://www.jointjs.com/demos/fixed-connection-points

**Reply by Naveena1321** [reply]
My problem with this is I can't directly use it on my standard shapes and may need to create custom shapes for all the shapes in stencills

**Reply by kumilingus** [reply, marked as answer]
There is no need to create custom shapes. You just need an equivalent function to `getLinkAnchors()` that will return an array of anchors for arbitrary shape. 

```ts
function getLinkAnchors(el) {
    switch (el.get('type')) {
        case 'standard.Rectangle':
           return [/* rectangle anchors */];
        case 'standard.Ellipse':
           return [/* ellipse anchors */];
         /* ... rest of the cases */         
    }
}
```
